packages/assets-controllers/src/AccountTrackerController.test.ts

```diff
@@ -139,91 +89,465 @@ describe('AccountTrackerController', () => {
+            '0x1': {
+              bar: { balance: '0x1' },
+              baz: { balance: '0x0' },
+            },
+            '0x2': {
+              bar: { balance: '0xa' },
+              baz: { balance: '0x0' },
+            },
+            '0x5': {
+              bar: { balance: '0x0' },
+              baz: { balance: '0x0' },
+            },
+          },
+        });
+      });
+
+      it('should get real balance', async () => {
+        mockedQuery.mockReturnValueOnce(Promise.resolve('0x10'));
+
+        const controller = new AccountTrackerController({
+          onPreferencesStateChange: sinon.stub(),
+          getIdentities: () => {
+            return { [ADDRESS_1]: {} as ContactEntry };
+          },
+          getSelectedAddress: () => ADDRESS_1,
+          getMultiAccountBalancesEnabled: () => true,
+          getCurrentChainId: () => '0x1',
+          getNetworkClientById: jest.fn().mockReturnValue({
+            configuration: {
+              chainId: '0x5',
+            },
+            provider,
+          }),
+        });
+
+        await controller.refresh('networkClientId1');
+
+        expect(controller.state).toStrictEqual({
+          accounts: {
+            [ADDRESS_1]: {
+              balance: '0x0',
+            },
+          },
+          accountsByChainId: {
+            '0x1': {
+              [ADDRESS_1]: {
+                balance: '0x0',
+              },
+            },
+            '0x5': {
+              [ADDRESS_1]: {
+                balance: '0x10',
+              },
+            },
+          },
+        });
+      });
+
+      it('should update only selected address balance when multi-account is disabled', async () => {
+        mockedQuery.mockReturnValueOnce(Promise.resolve('0x10'));
+
+        const controller = new AccountTrackerController({
+          onPreferencesStateChange: sinon.stub(),
+          getIdentities: () => {
+            return {
+              [ADDRESS_1]: {} as ContactEntry,
+              [ADDRESS_2]: {} as ContactEntry,
+            };
+          },
+          getSelectedAddress: () => ADDRESS_1,
+          getMultiAccountBalancesEnabled: () => false,
+          getCurrentChainId: () => '0x1',
+          getNetworkClientById: jest.fn().mockReturnValue({
+            configuration: {
+              chainId: '0x5',
+            },
+            provider,
+          }),
+        });
+
+        await controller.refresh('networkClientId1');
+
+        expect(controller.state).toStrictEqual({
+          accounts: {
+            [ADDRESS_1]: { balance: '0x0' },
+            [ADDRESS_2]: { balance: '0x0' },
+          },
+          accountsByChainId: {
+            '0x1': {
+              [ADDRESS_1]: { balance: '0x0' },
+              [ADDRESS_2]: { balance: '0x0' },
+            },
+            '0x5': {
+              [ADDRESS_1]: { balance: '0x10' },
+              [ADDRESS_2]: { balance: '0x0' },
+            },
+          },
+        });
+      });
```

Comment:
just curious, with the way the test is set up would the result be different if `getMultiAccountBalancesEnabled` returned true?

Reply:
Ah I see the next test sortof answers it. Maybe there needs to be a second `mockedQuery.mockReturnValueOnce(...)` to sortof actually test the thing its claiming to test.

Reply:
https://github.com/MetaMask/core/pull/3586/commits/af38a0f11df23011a1ebcae8aee20493e14bfc39